x/superfluid/keeper/stake.go

```diff
@@ -269,23 +269,108 @@ func (k Keeper) SuperfluidUndelegate(ctx sdk.Context, sender string, lockID uint
 // SuperfluidUnbondLock unbonds the lock that has been used for superfluid staking.
 // This method would return an error if the underlying lock is not superfluid undelegating.
 func (k Keeper) SuperfluidUnbondLock(ctx sdk.Context, underlyingLockId uint64, sender string) error {
+	_, err := k.unbondLock(ctx, underlyingLockId, sender, sdk.Coins{})
+	return err
```

Comment:
nit: just wondering if there's a specific reason why you didn't do this here instead a declaring a whole new function? 

```suggestion
		lock, err := k.lk.GetLockByID(ctx, underlyingLockId)
	if err != nil {
		return 0, err
	}
	err = k.validateLockForSF(ctx, lock, sender)
	if err != nil {
		return 0, err
	}
	synthLocks := k.lk.GetAllSyntheticLockupsByLockup(ctx, underlyingLockId)
	if len(synthLocks) != 1 {
		return 0, types.ErrNotSuperfluidUsedLockup
	}
	if !synthLocks[0].IsUnlocking() {
		return 0, types.ErrBondingLockupNotSupported
	}
	return k.lk.BeginForceUnlock(ctx, underlyingLockId, coins)
}
```

Reply:
It's because the function `unbondLock` is also used in `SuperfluidUndelegateAndUnbondLock`
https://github.com/stakewithus/osmosis/blob/3930891ded8c4f7ab10b0f854a9368bcdaeaa997/x/superfluid/keeper/stake.go#L308